src/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverter.java

```diff
@@ -74,6 +79,25 @@ public SwaggerConverter(Scheme defaultScheme, String defn, ValueGenerator valueG
     private List<OperationModel> readOpenAPISpec() throws SwaggerException {
         List<OperationModel> operations = new LinkedList<>();
         Swagger swagger = new SwaggerParser().parse(this.defn);
+        
+        if (swagger == null) {
+            try {
+                // Try the older spec
+                // Annoyingly the converter only reads files
+                File temp = File.createTempFile("openapi", ".defn");
+                BufferedWriter bw = new BufferedWriter(new FileWriter(temp));
+                bw.write(this.defn);
+                bw.close();
+                
+                swagger = new SwaggerCompatConverter().read(temp.getAbsolutePath());
+                if (!temp.delete()) {
+                    LOG.debug("Failed to delete " + temp.getAbsolutePath());

```

Comment:
Shouldn't this be a warn or error?

If a user repeatedly tries to parse something, and temp files keep getting created and not removed it could lead to a storage issue. If it's left as debug is should probably be wrapped with an if.